jax/__init__.py

```diff
@@ -175,6 +175,54 @@
 from jax._src.array import Shard as Shard
 
 
+_deprecations = {
+  # Added 28 March 2023
+  "ShapedArray": (
+    "jax.ShapedArray is deprecated. Use jax.core.ShapedArray",
+    _deprecated_ShapedArray,
+  ),
+  "ad": (
+    "jax.ad is deprecated. Use jax.interpreters.ad",
+    _deprecated_ad,
+  ),
+  "partial_eval": (
+    "jax.partial_eval is deprecated. Use jax.interpreters.partial_eval",
+    _deprecated_partial_eval,
+  ),
+  "pxla": (
+    "jax.pxla is deprecated. Use jax.interpreters.pxla",
+    _deprecated_pxla,
+  ),
+  "xla": (
+    "jax.ad is deprecated. Use jax.interpreters.xla",
```

Comment:
drive-by

nit: should this be jax.xla?

Reply:
Fixed, thanks